### ISSUE TITLE
Armory Scenario Not Finding Torch Module While Using Custom Model

### ISSUE THREAD

**Issue by dominic-simon** [twosixlabs/armory#1396]
I'm trying to write a custom model for a scenario in which I use the torch module. However, I get a ModuleNotFoundError saying there is no module named torch. I have no idea why since I can run normal python scripts with torch just fine. When I remove torch and all its references, I get the same error for torchvision. Below is everything I have from my command line. If there's anything I'm doing incorrectly or if there's more information needed, please let me know. Any help with solving this is appreciated.
![Screenshot from 2022-04-06 16-40-26](https://user-images.githubusercontent.com/84974463/162067893-c0adaf7e-502f-4a6f-8665-570a9045e1a6.png)

**Comment by davidslater** [comment]
Can you show the command you used to launch `armory` and your json config as well?

**Comment by dominic-simon** [comment]
**Commands:**
cd /home/dominicsimon/Desktop/work/algorithm/
armory run /home/dominicsimon/Desktop/work/algorithm/resisc45_baseline_densenet121_patch.json

**json file**
```
{
    "_description": "Resisc45 image classification, contributed by MITRE Corporation",
    "adhoc": null,
    "attack": {
        "knowledge": "white",
        "kwargs": {
            "apply_patch_args": [],
            "apply_patch_kwargs": {
                "scale": 0.3
            },
            "batch_size": 1,
            "learning_rate": 5.0,
            "max_iter": 3,
            "rotation_max": 22.5,
            "scale_max": 0.3,
            "scale_min": 0.29999,
            "targeted": true
        },
        "module": "art.attacks.evasion",
        "name": "AdversarialPatch",
        "targeted_labels": {
            "kwargs": {
                "num_classes": 45
            },
            "module": "armory.utils.labels",
            "name": "RoundRobinTargeter"
        },
        "type": "patch",
        "use_label": true
    },
    "dataset": {
        "batch_size": 1,
        "framework": "numpy",
        "module": "armory.data.datasets",
        "name": "resisc45"
    },
    "defense": null,
    "metric": {
        "means": true,
        "perturbation": "image_circle_patch_diameter",
        "record_metric_per_sample": false,
        "task": [
            "categorical_accuracy"
        ]
    },
    "model": {
        "fit": false,
        "fit_kwargs": {},
        "model_kwargs": {},
        "module": "armory_script",
        "name": "run_model",
        "weights_file": "null",
        "wrapper_kwargs": {}
    },
    "scenario": {
        "kwargs": {},
        "module": "armory.scenarios.image_classification",
        "name": "ImageClassificationTask"
    },
    "sysconfig": {
        "docker_image": "twosixarmory/tf1:0.14.4",
        "external_github_repo": null,
        "gpus": "all",
        "output_dir": null,
        "output_filename": null,
        "use_gpu": false
    }
}

Thank you for your help!

**Comment by davidslater** [comment]
If your model is pytorch based, you need to use the pytorch docker image. Change your config from `twosixarmory/tf1:0.14.4` to `twosixarmory/pytorch:0.14.4`

**Comment by davidslater** [comment]
Or, if you have torch installed locally, try doing `armory run /home/dominicsimon/Desktop/work/algorithm/resisc45_baseline_densenet121_patch.json --no-docker`
